synth-converter/src/graph/graph_builder.rs

```diff
@@ -0,0 +1,359 @@
+
+        if let Some(temperature_shaker) = &action.temperature_shaker {
+            self.insert_an_observation(
+                &action_term,
+                &cat::temperatureShakerShape,
+                temperature_shaker,
+            )
+            .context("Failed to insert observation")?
+        }
+
+        if let Some(temperature_tumble_stirrer) = &action.temperature_tumble_stirrer {
+            self.insert_an_observation(
+                &action_term,
+                &cat::temperatureTumbleStirrerShape,
+                temperature_tumble_stirrer,
+            )
+            .context("Failed to insert observation")?
+        }
+
+        if let Some(speed_shaker) = &action.speed_shaker {
+            self.insert_an_observation(&action_term, &cat::speedInRPM, speed_shaker)?;
+        }
+
+        if let Some(dispense_type) = &action.dispense_type {
+            self.graph
+                .insert(&action_term, cat::dispenseType, dispense_type.as_str())?;
+        }
+
+        if let Some(dispense_state) = &action.dispense_state {
+            self.graph
+                .insert(&action_term, alloqual::AFQ_0000111, dispense_state.as_str())?;
+        }
+
+        if let Some(container_positions) = &action.has_container_position_and_quantity {
+            for container_position in container_positions {
+                self.insert_a_container_position(&action_term, container_position)?;
+            }
+        }
+
+        if let Some(sample) = &action.has_sample {
+            self.insert_samples(&action_term, sample)?;
+        }
+
+        self.insert_action_type(&action_term, action)?;
+
+        Ok(())
+    }
+
+    /// Insert a batch
+    ///
+    /// Assumes a new graph has been created
+    ///
+    /// # Returns
+    /// A `Result` containing () if successful, or an error if the graph building fails.
+    pub fn insert_a_batch(&mut self, batch: &Batch) -> Result<()> {
+        let batch_term = generate_bnode_term();
+
+        self.graph.insert(&batch_term, rdf::type_, cat::Batch)?;
+        self.graph
+            .insert(&batch_term, schema::name, batch.batch_id.as_str())?;
+
+        // Propagate any error returned from `insert_an_action`
+        for action in &batch.actions {
+            self.insert_an_action(&batch_term, action)?;
+        }
+
+        Ok(())
+    }
+
+    /// Get the turtle serialization of the RDF graph
+    ///
+    /// Assumes a new graph has been created and built.
+    ///
+    /// # Returns
+    /// A `Result` containing the graph as Turtle serialization, or an error
+    /// if the graph retrieval fails.
+    pub fn serialize_to_turtle(&self) -> Result<String> {
+        serialize_graph_to_turtle(&self.graph).context("Failed to serialize graph to Turtle")
+    }
+
+    /// Get the turtle serialization of the RDF graph
+    ///
+    /// Assumes a new graph has been created and built.
+    ///
+    /// # Returns
+    /// A `Result` containing the graph as jsonld serialization, or an error
```

Comment:
suggestion: You can keep it more brief, dito to the others...: 

```suggestion
    ///  The `jsonld` serialization of the grap, or an error otherwise.
```